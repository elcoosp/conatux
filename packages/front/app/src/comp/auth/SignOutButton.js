import React from 'react'

import Auth from '@:/services/auth'
import Buttons from 'ui/buttons'

const SignOutButton = () => (
  <Buttons.Basic onClick={Auth.signOut}>Sign out</Buttons.Basic>
)
export default SignOutButton
