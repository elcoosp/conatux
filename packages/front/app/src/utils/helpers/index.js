export const pipe = {
  execAsync: (...funcs) => context =>
    funcs.reduce(async (ctx, func) => {
      await func(ctx)
      return ctx
    }, context)
}
