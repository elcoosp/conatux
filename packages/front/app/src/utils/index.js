export { default as testing } from './testing'
export { default as linkify } from './linkify'
