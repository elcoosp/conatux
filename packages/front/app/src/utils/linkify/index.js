import React from 'react'
import { Link } from 'react-router-dom'

export const labelize = (path, label) => ({
  path,
  label
})

export const fromRoutes = (routesTree, componentName) => {
  if (['label', 'path'].every(prop => routesTree.hasOwnProperty(prop))) {
    const { path, label } = routesTree
    const link = props => (
      <Link to={path} {...props}>
        {label}
      </Link>
    )
    link.displayName = componentName
    return link
  }

  return Object.entries(routesTree).reduce(
    (links, [routeName, nestedRoute]) => {
      const componentName =
        routeName.slice(0, 1).toUpperCase() + routeName.slice(1)
      links[componentName] = fromRoutes(nestedRoute, componentName)
      return links
    },
    {}
  )
}
