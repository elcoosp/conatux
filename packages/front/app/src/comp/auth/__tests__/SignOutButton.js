import React from 'react'
import { desc } from '@:/utils/testing'
import Auth from '@:/services/auth'
import { shallow } from 'enzyme'
import SignOutButton from '../SignOutButton'

const props = {}

desc.compWithSnap({ SignOutButton }, props, () => {
  it('should call `Auth.signOut()` on click', () => {
    Auth.signOut = jest.fn()
    const wrapper = shallow(<SignOutButton />)
    wrapper.simulate('click')
    expect(Auth.signOut).toHaveBeenCalledTimes(1)
  })
})
