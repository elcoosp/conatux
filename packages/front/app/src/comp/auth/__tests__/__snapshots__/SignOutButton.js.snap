// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignOutButton /> should match snapshot 1`] = `
<styled.button
  onClick={[Function]}
>
  Sign out
</styled.button>
`;
